catroid/src/main/java/org/catrobat/catroid/ui/MultilingualActivity.java

```diff
@@ -0,0 +1,120 @@
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.util.DisplayMetrics;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import org.catrobat.catroid.R;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static org.catrobat.catroid.CatroidApplication.defaultSystemLanguage;
+import static org.catrobat.catroid.CatroidApplication.languageSharedPreferences;
+import static org.catrobat.catroid.common.Constants.LANGUAGE_CODE;
+import static org.catrobat.catroid.common.Constants.LANGUAGE_TAG_KEY;
+
+public class MultilingualActivity extends Activity {
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_multilingual);
+		setTitle(R.string.preference_title_language);
+		ListView listview = (ListView) findViewById(R.id.list_languages);
+		final List<String> languagesNames = new ArrayList<>();
+		languagesNames.add(getResources().getString(R.string.device_language));
+		for (String aLanguageCode : LANGUAGE_CODE) {
+			if (aLanguageCode.length() == 2) {
+				languagesNames.add(new Locale(aLanguageCode).getDisplayName(new Locale(aLanguageCode)));
+			} else {
+				String language = aLanguageCode.substring(0, 2);
+				String country = aLanguageCode.substring(4);
+				languagesNames.add(new Locale(language, country).getDisplayName(new Locale(language, country)));
+			}
+		}
+		ArrayAdapter<String> adapter = new ArrayAdapter<>(this, R.layout.multilingual_name_text, R.id.lang_text, languagesNames);
+		listview.setAdapter(adapter);
+		listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+			@Override
+			public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+
+				if (position == 0) {
+					setNewLocale(defaultSystemLanguage, null);
+				} else if (LANGUAGE_CODE[position - 1].length() == 2) {
```

Comment:
Would it be possible to make use of Andropi's `Locale` here?
https://developer.android.com/reference/java/util/Locale.html

Reply:
could you please be more specific?I didn't understand your question!
` defaultSystemLanguage = Locale.getDefault().getLanguage();` which defined in CatroidApplication!

Reply:
sorry, I just looked it up again, I misunderstood something here, never mind ;)

Reply:
no problem, if you have further questions, I can answer them today after the Meeting if you have time.